install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -894,3 +891,17 @@ data:
 {{- end }}
 
 {{- end }}
+
+---
+{{- if and .Values.ipMasqAgent .Values.ipMasqAgent.enabled }}
```

Comment:
What will happen if the ip-masq-agent ConfigMap already exist? Won't it get overriden which might be unexpected by users? I think we should extend the if-condition to check whether `.Values.ipMasqAgent.config` is set.

Reply:
> I'm a bit confused what the core proposal is here. I suspect we just need some documentation about how to use the feature, not Helm changes.
> 
> I believe you should be able to do:
> 
> ```
> helm install ... --namespace kube-system --set ipMasqAgent.enabled=true
> ```
> 
> And then create a new ConfigMap in `kube-system` namespace with the complete ipmasq agent configuration, whatever this is generating:
> 
> ```
> apiVersion: v1
> kind: ConfigMap
> metadata:
>   name: ip-masq-agent
>   namespace: {{ .Release.Namespace }}
> data:
>   enable-ip-masq-agent: "true"
>   config: |-
> {{ toJson .Values.ipMasqAgent.config | indent 4 }}
> {{- end }}
> ```
> 
> If we put this into the Helm charts, then my concern is that users who use this feature and upgrade will have their settings overwritten. I'm not sure if it even makes sense for Cilium's Helm configuration to control this configmap?

@joestringer  Sorry to too long time no reply! Refer to @alphabet5 's comments.  I think his approach avoids the problem of  configuration override.  leaving `.Values.ipMasqAgent.config` commented out by default. 

When install  with `--set ipMasqAgent.enabled=true --set ipMasqAgent.config.masqLinkLocal=true --set "ipMasqAgent.config.nonMasqueradeCIDRs={1.1.1.1/32 ,2.2.2.2/32}" ,It will creates the configMap, which will not be overwritten even if the user updates.